src/components/Farmhand/Farmhand.js

```diff
@@ -420,7 +421,7 @@ export default class Farmhand extends FarmhandReducers {
   }
 
   get isForestUnlocked() {
-    return levelAchieved(this.state.experience) >= UNLOCK_FOREST_LEVEL
+    return this.state.stagesUnlocked[stageFocusType.FOREST]
```

Comment:
It seems like we might be able to avoid adding the `stagesUnlocked` state by computing the level entitlements. I'm thinking something like this:

```patch
diff --git a/src/components/Farmhand/Farmhand.js b/src/components/Farmhand/Farmhand.js
index 5859e00a..dace1f6e 100644
--- a/src/components/Farmhand/Farmhand.js
+++ b/src/components/Farmhand/Farmhand.js
@@ -421,7 +421,7 @@ export default class Farmhand extends FarmhandReducers {
   }
 
   get isForestUnlocked() {
-    return this.state.stagesUnlocked[stageFocusType.FOREST]
+    return this.levelEntitlements.stageFocusType === true
   }
 
   /**
diff --git a/src/index.js b/src/index.js
index a969917e..ff12417d 100644
--- a/src/index.js
+++ b/src/index.js
@@ -236,6 +236,7 @@
  * @property {number} sprinklerRange
  * @property {Object.<string, boolean>} items
  * @property {Object.<string, boolean>} tools
+ * @property {Object.<string, boolean>} stageFocusType
  */
 
 /**
diff --git a/src/utils/getLevelEntitlements.js b/src/utils/getLevelEntitlements.js
index 15a49daa..b1152d4d 100644
--- a/src/utils/getLevelEntitlements.js
+++ b/src/utils/getLevelEntitlements.js
@@ -20,11 +20,12 @@ export const getLevelEntitlements = memoize(
       sprinklerRange: INITIAL_SPRINKLER_RANGE,
       items: {},
       tools: {},
+      stageFocusType: {},
     }
 
     // Assumes that levels is sorted by id.
     levels.find(
-      ({ unlocksShopItem, unlocksTool, id, increasesSprinklerRange }) => {
+      ({ unlocksShopItem, unlocksTool, unlocksStageFocusType, id, increasesSprinklerRange }) => {
         if (increasesSprinklerRange) {
           acc.sprinklerRange++
         }
@@ -37,6 +38,10 @@ export const getLevelEntitlements = memoize(
           acc.tools[unlocksTool] = true
         }
 
+        if (unlocksStageFocusType) {
+          acc.stageFocusType[unlocksStageFocusType] = true
+        }
+
         return id === levelNumber
       }
     )
diff --git a/src/utils/getLevelEntitlements.test.js b/src/utils/getLevelEntitlements.test.js
index 39021fbd..1f267b09 100644
--- a/src/utils/getLevelEntitlements.test.js
+++ b/src/utils/getLevelEntitlements.test.js
@@ -16,6 +16,7 @@ describe('getLevelEntitlements', () => {
       tools: {
         SHOVEL: true,
       },
+      stageFocusType: {},
     })
   })
 })
```

What do you think @lstebner?

Reply:
nice one, just made this change! everything seems to be working so I think I got it all right, but definitely let me know if something seems off or missed